test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ERCPrecompileSuite.java

```diff
@@ -1240,6 +1244,290 @@ private HapiApiSpec getErc721OwnerOfFromErc20TokenFails() {
+																.status(SUCCESS)
+																.contractCallResult(
+																		resultWith()
+																				.contractCallResult(htsPrecompileResult()
+																						.forFunction(HTSPrecompileResult.FunctionType.ERC_TRANSFER)
+																						.withErcFungibleTransferStatus(true)
+																				)
+																)
+												)))
+				);
+	}
+
+	private HapiApiSpec directCallsWorkForERC721() {
+
+		final AtomicReference<String> tokenNum = new AtomicReference<>();
+
+		final var tokenName = "TokenA";
+		final var tokenSymbol = "FDFDFD";
+		final var tokenTotalSupply = 1;
+
+		final var symbolTxn = "symbolTxn";
+		final var nameTxn = "nameTxn";
+		final var tokenURITxn = "tokenURITxn";
+		final var totalSupplyTxn = "totalSupplyTxn";
+		final var balanceOfTxn = "balanceOfTxn";
+		final var ownerOfTxn = "ownerOfTxn";
+
+		return defaultHapiSpec("DirectCallsWorkForERC721")
+				.given(
+						newKeyNamed(MULTI_KEY),
+						cryptoCreate(ACCOUNT).balance(100 * ONE_HUNDRED_HBARS),
+						cryptoCreate(RECIPIENT).balance(100 * ONE_HUNDRED_HBARS),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(NON_FUNGIBLE_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.name(tokenName)
+								.symbol(tokenSymbol)
+								.initialSupply(0)
+								.treasury(TOKEN_TREASURY)
+								.adminKey(MULTI_KEY)
+								.supplyKey(MULTI_KEY)
+								.exposingCreatedIdTo(tokenNum::set),
+						mintToken(NON_FUNGIBLE_TOKEN, List.of(FIRST_META)),
+						tokenAssociate(ACCOUNT, NON_FUNGIBLE_TOKEN),
+						tokenAssociate(RECIPIENT, NON_FUNGIBLE_TOKEN),
+						cryptoTransfer(TokenMovement.movingUnique(NON_FUNGIBLE_TOKEN, 1).between(TOKEN_TREASURY,
+								ACCOUNT))
+				).when(withOpContext(
+						(spec, ignore) -> {
+							var tokenAddress = asHexedSolidityAddress(asToken(tokenNum.get()));
+							allRunFor(spec,
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "name", "ERC721ABI")
+									).via(nameTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "symbol", "ERC721ABI")
+									).via(symbolTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "tokenURI", "ERC721ABI"),
+											1
+									).via(tokenURITxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "totalSupply", "ERC721ABI")
+									).via(totalSupplyTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "balanceOf", "ERC721ABI"),
+											asHexedSolidityAddress(spec.registry().getAccountID(ACCOUNT))
+									).via(balanceOfTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "ownerOf", "ERC721ABI"),
+											1
+									).via(ownerOfTxn)
+							);
```

Comment:
"ERC721ABI" can be extracted in a single variable. If you used ERC721ABI contract, haven't you missed uploading the .sol, .bin and .json files?

Reply:
There is no ERC721ABI contract, I just use the ABI for standard ERC721 functions and added the .json file to make use of the `getABIFor` and `contractCallWithFunctionAbi()` methods.